google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/data/v2/stub/DynamicFlowControlCallableTest.java

```diff
@@ -277,32 +281,30 @@ public ApiFuture<List<MutateRowsResponse>> futureCall(
 
   private void createFlowControlEvent(final FlowController flowController) throws Exception {
     flowController.reserve(INITIAL_ELEMENT, 0);
-    final AtomicBoolean threadStarted = new AtomicBoolean(false);
+    CompletableFuture<Void> threadStarted = new CompletableFuture<>();
+    CompletableFuture<Void> threadReservedOne = new CompletableFuture<>();
     Thread t =
         new Thread(
             new Runnable() {
               @Override
               public void run() {
+                threadStarted.complete(null);
                 try {
-                  threadStarted.set(true);
                   flowController.reserve(1, 0);
+                  threadReservedOne.complete(null);
                 } catch (Exception e) {
+                  threadReservedOne.completeExceptionally(e);
                 }
               }
             });
     t.start();
     // Wait 5 seconds for the thread to start, and 50 milliseconds after it's started to make sure
```

Comment:
nit: I think we can remove "5 seconds for the thread to start, and" from the comment now.

Reply:
fixed